### PR TITLE
WMS-150 | Add NavigationContainer

### PR DESCRIPTION
**Pull request by PedroHLC**
Objective:
* Follow [Sufiyan’s design pattern](https://www.sketch.com/s/a252928f-582c-4f38-a5ee-14d60a2a3067/a/zaaeQy) guidelines
* Follow Elm’s with-pattern API guidelines
* Develop one component capable to adapt as:
  * [X] a Desktop's SideBar capable of toggling width size;
  * [x] a Mobile's Header with Sandwith Button;
  * [x] a Mobile's Drawer (hidden SideBar).

### CODE REVIEW

File: src/UI/Internal/SideBar.elm [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+pagesView cfg pages navExpanded =
+    let
+        item { labeledIcon, link, isCurrent } =
+            if navExpanded then
+                pageItem cfg labeledIcon link isCurrent
+
+            else
+                slimPageItem cfg labeledIcon link isCurrent
+
+        pad =
```
Comment (jouderianjr) [inline src/UI/Internal/SideBar.elm:142]: maybe `paddingAttr`

File: src/UI/Internal/SideBar.elm [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                    { top = 0
+                    , left = 4
+                    , right = 7
+                    , bottom = 0
+                    }
+
+            else
+                padding 0
+
+        space =
```
Comment (jouderianjr) [inline src/UI/Internal/SideBar.elm:154]: maybe `spacingAttr`

File: src/UI/Internal/SideBar.elm [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+-- Render
+
+
+desktopColumn : RenderConfig -> Element msg -> Menu msg -> Element msg
+desktopColumn cfg page sidebar =
+    Element.row [ width fill, height fill ]
+        [ view cfg sidebar
+        , Element.column
+            [ width fill
+            , Element.vhHeight 100
```
[the remark this answers is not in the document]
Comment (PedroHLC) [reply src/UI/Internal/SideBar.elm:28]: Sadly no, `height fill` would make the sidebar as long as the other column.

File: src/UI/Internal/SideBar.elm [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+            , width fill
+            , height (px (60 + 96))
+            ]
+
+        logoEl =
+            case logo of
+                Just { hint, body } ->
+                    [ body ]
+                        -- TODO: Add Hint
+                        |> Element.column
```
[the remark this answers is not in the document]
Comment (PedroHLC) [reply src/UI/Internal/SideBar.elm:93]: I did, but that didn't play well with elements that didn't rely well on flex-basis (aka Event's grid)

**Pull request merged**